src/utils.ts

```diff
@@ -1,18 +1,24 @@
 import type {Component, RefObject, ComponentClass} from 'react';
 import {Image, findNodeHandle} from 'react-native';
-import type {ImageSourcePropType} from 'react-native';
 import type {ReactVideoSource, ReactVideoSourceProperties} from './types/video';
 
-type Source = ImageSourcePropType | ReactVideoSource;
-
 export function resolveAssetSourceForVideo(
-  source: Source,
+  source: ReactVideoSource,
 ): ReactVideoSourceProperties {
+  // This is deprecated, but we need to support it for backward compatibility
   if (typeof source === 'number') {
     return {
       uri: Image.resolveAssetSource(source).uri,
     };
   }
+
+  if (typeof source.uri === 'number') {
+    return {
+      ...source,
+      uri: Image.resolveAssetSource(source.uri).uri,
+    };
+  }
+
```

Comment:
@freeboub In stand of making breaking change I will leave old option and add reference in docs to use new way. With this we won't have to make breaking change

> No native side change ?

Yes as you can see, we earlier converted `require` into `{ uri: "some uri to local file" }` so basically I've just added one if statement